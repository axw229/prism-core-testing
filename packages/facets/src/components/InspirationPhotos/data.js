// @flow
import tinycolor from '@ctrl/tinycolor'
import { type Color } from 'src/shared/types/Colors.js.flow'
import uniqueId from 'lodash/uniqueId'

export const throttleDragTime = 5
// we need this two const to make sure cursor always point to the center of preview circle
export const activedPinsHalfWidth = 24

export const cloneColorPinsArr = (pins) => {
  return pins.map(pin => ({ ...pin }))
}

export const getRGBInitPins = (acgColors: Array<Object>) => {
  return acgColors.map<Object>((acgColor: any): Object => {
    return {
      rgbArray: [acgColor.r, acgColor.g, acgColor.b],
      rgbValue: `rgb(${acgColor.r},${acgColor.g},${acgColor.b})`,
      translateValueX: acgColor.x,
      translateValueY: acgColor.y
    }
  })
}

export const rgb2lab = (rgb: Array<any>) => {
  let r = rgb[0] / 255
  let g = rgb[1] / 255
  let b = rgb[2] / 255
  let x; let y; let z

  r = (r > 0.04045) ? Math.pow((r + 0.055) / 1.055, 2.4) : r / 12.92
  g = (g > 0.04045) ? Math.pow((g + 0.055) / 1.055, 2.4) : g / 12.92
  b = (b > 0.04045) ? Math.pow((b + 0.055) / 1.055, 2.4) : b / 12.92

  x = (r * 0.4124 + g * 0.3576 + b * 0.1805) / 0.95047
  y = (r * 0.2126 + g * 0.7152 + b * 0.0722) / 1.00000
  z = (r * 0.0193 + g * 0.1192 + b * 0.9505) / 1.08883

  x = (x > 0.008856) ? Math.pow(x, 1 / 3) : (7.787 * x) + 16 / 116
  y = (y > 0.008856) ? Math.pow(y, 1 / 3) : (7.787 * y) + 16 / 116
  z = (z > 0.008856) ? Math.pow(z, 1 / 3) : (7.787 * z) + 16 / 116

  return [(116 * y) - 16, 500 * (x - y), 200 * (y - z)]
}

export type RenderingPin = {
  isActiveFlag: boolean,
  name: string,
  colorNumber: number,
  rgbValue: string,
  pinNumber: number,
  isContentLeft: boolean,
  translateX: number,
  translateValueY: number,
  pinId: string
}

export const renderingPins = (initPins: Array<Object>, canvasOffsetWidth: number, canvasOffsetHeight: number, colors: Color[]): RenderingPin[] => {
  return getRGBInitPins(initPins).map<Object>((acgColor: any, index: number): Object => {
    const color: Color = findClosestColor(acgColor.rgbArray, colors)
    const calculateTranslateX = acgColor.translateValueX * canvasOffsetWidth
    const calculateTranslateY = acgColor.translateValueY * canvasOffsetHeight

    return {
      ...color,
      isActiveFlag: (initPins.length - 1 === index),
      rgbValue: `rgb(${color.red},${color.green},${color.blue})`,
      pinNumber: index,
      pinId: uniqueId('pin_'),
      isContentLeft: calculateTranslateX < canvasOffsetWidth / 2,
      translateX: Math.min(canvasOffsetWidth - 2 * activedPinsHalfWidth, calculateTranslateX),
      translateY: Math.min(canvasOffsetHeight - 2 * activedPinsHalfWidth, calculateTranslateY)
    }
  })
}

export const findClosestColor = ([red, green, blue]: [number, number, number], colors: Color[]): Color => {
  const currentPixelHex: string = tinycolor(`rgb (${red}, ${green}, ${blue})`).toHex()

  return colors.reduce(([closestColor, smallestDistanceSoFar], color): [Color, number] => {
    const colorDistance: number = measureDistance(currentPixelHex, tinycolor(`rgb(${color.red},${color.green},${color.blue})`).toHex())
    return colorDistance < smallestDistanceSoFar ? [color, colorDistance] : [closestColor, smallestDistanceSoFar]
  }, [colors[0], 99999])[0]
}

export const measureDistance = (color1: string, color2: string): number => {
  const position1 = getChromaLuma(color1)
  const position2 = getChromaLuma(color2)
  const redDelta = position1[0] - position2[0]
  const greenDelta = position1[1] - position2[1]
  const lumaDelta = position1[2] - position2[2]
  const distance = redDelta * redDelta + greenDelta * greenDelta + lumaDelta * lumaDelta
  return distance
}

export const getChromaLuma = (rgbString: string) => {
  const color = tinycolor(rgbString)
  const luminosity = color.getLuminance()
  const colorRgb = color.toRgb()
  const red = colorRgb.r
  const green = colorRgb.g
  const blue = colorRgb.b
  const sum = red + green + blue
  let redChroma
  let greenChroma
  if (sum > 25) {
    redChroma = red / sum
    greenChroma = green / sum
  } else {
    redChroma = 1 / 3
    greenChroma = 1 / 3
  }

  return [redChroma, greenChroma, luminosity]
}
