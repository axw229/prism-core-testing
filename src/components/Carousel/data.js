// This is temporary use for mock carousel data from api
// @flow
import type { Color } from '../types/Colors'
import * as Colors from '../../../__mocks__/data/color/Colors'
const colors = Colors.getAllColors()
const getExpertColors = (colors: Color, colorList: array) => {
  let data = []
  for (var i = 0; i < colorList.length; i = i + 3) {
    let tmp = []
    let j = 0
    while (j !== 3) {
      const color = Object.values(colors).find((color) => {
        return (colorList[i + j] === color.colorNumber)
      })
      tmp.push(color)
      j++
    }
    data.push(tmp)
  }
  return data
}
export const colorList = [
  '0071', '6039', '0053', '6232', '0041', '9171',
  '6496', '7063', '6747', '7579', '2739', '7502',
  '7701', '9174', '7636', '0063', '7006', '6808',
  '7622', '8917', '9110', '9178', '7538', '7578',
  '6732', '7000', '6797', '7623', '6039', '9179',
  '6224', '9175', '0034', '6839', '6071', '6656',
  '6774', '7675', '9101', '7568', '7048', '6044',
  '6424', '6573', '6680', '7592', '6244', '0037',
  '6265', '7613', '7577', '6557', '6169', '6720',
  '2844', '6221', '7104', '7001', '6650', '6967',
  '0007', '7076', '6222', '0064', '7030', '6263',
  '6052', '7566', '7666', '6788', '6841', '6156',
  '0077', '6258', '7647', '7612', '7697', '7057',
  '6510', '6131', '6199', '6464', '6556', '6771',
  '6921', '6859', '7018', '6963', '6711', '7036',
  '0073', '6463', '6353', '6164', '0012', '7674',
  '6840', '7602', '6109', '6988', '6768', '7006',
  '6759', '6344', '6802', '6349', '6020', '6307',
  '6545', '6417', '7592', '6026', '6117', '6467',
  '6247', '7031', '6155', '6979', '6725', '6963',
  '6479', '0071', '6249', '6607', '6107', '7083',
  '6018', '7058', '6256', '0056', '7662', '7135',
  '6265', '7071', '6010', '0015', '6113', '7604'
]

export const allCollectionsData = [
  {
    tabId: 'tab1',
    tabName: 'MOST POPULAR',
    collections: [
      {
        id: '1',
        name: '2019 Aficionado',
        img: 'https://www.sherwin-williams.com/dist/cvt/assets/images/collections/2019-shapeshifter-thumb.jpg',
        colorCollections: ['7579', '2739', '6404', '6407', '7705', '7502', '2704']
      },
      {
        id: '2',
        name: '2019 Naturalist',
        img: 'https://www.sherwin-williams.com/dist/cvt/assets/images/collections/2019-shapeshifter-thumb.jpg',
        colorCollections: ['6289', '9031', '6860', '9171', '6232', '9173', '0041']
      },
      {
        id: '3',
        name: '2019 Naturalist',
        img: 'https://www.sherwin-williams.com/dist/cvt/assets/images/collections/2019-shapeshifter-thumb.jpg',
        colorCollections: ['6289', '9031', '6860', '9171', '6232', '9173', '0041']
      },
      {
        id: '3',
        name: '2019 Naturalist',
        img: 'https://www.sherwin-williams.com/dist/cvt/assets/images/collections/2019-shapeshifter-thumb.jpg',
        colorCollections: ['6289', '9031', '6860', '9171', '6232', '9173', '0041']
      },
      {
        id: '3',
        name: '2019 Naturalist',
        img: 'https://www.sherwin-williams.com/dist/cvt/assets/images/collections/2019-shapeshifter-thumb.jpg',
        colorCollections: ['6289', '9031', '6860', '9171', '6232', '9173', '0041']
      },
      {
        id: '3',
        name: '2019 Naturalist',
        img: 'https://www.sherwin-williams.com/dist/cvt/assets/images/collections/2019-shapeshifter-thumb.jpg',
        colorCollections: ['6289', '9031', '6860', '9171', '6232', '9173', '0041']
      },
      {
        id: '3',
        name: '2019 Naturalist',
        img: 'https://www.sherwin-williams.com/dist/cvt/assets/images/collections/2019-shapeshifter-thumb.jpg',
        colorCollections: ['6289', '9031', '6860', '9171', '6232', '9173', '0041']
      },
      {
        id: '3',
        name: '2019 Naturalist',
        img: 'https://www.sherwin-williams.com/dist/cvt/assets/images/collections/2019-shapeshifter-thumb.jpg',
        colorCollections: ['6289', '9031', '6860', '9171', '6232', '9173', '0041']
      },
      {
        id: '3',
        name: '2019 Naturalist',
        img: 'https://www.sherwin-williams.com/dist/cvt/assets/images/collections/2019-shapeshifter-thumb.jpg',
        colorCollections: ['6289', '9031', '6860', '9171', '6232', '9173', '0041']
      }, {
        id: '3',
        name: '2019 Naturalist',
        img: 'https://www.sherwin-williams.com/dist/cvt/assets/images/collections/2019-shapeshifter-thumb.jpg',
        colorCollections: ['6289', '9031', '6860', '9171', '6232', '9173', '0041']
      },
      {
        id: '3',
        name: '2019 Naturalist',
        img: 'https://www.sherwin-williams.com/dist/cvt/assets/images/collections/2019-shapeshifter-thumb.jpg',
        colorCollections: ['6289', '9031', '6860', '9171', '6232', '9173', '0041']
      }
    ]
  },
  {
    tabId: 'tab2',
    tabName: 'COLOR ID',
    collections: [
      {
        id: '1',
        name: 'Free Spirit',
        img: 'https://www.sherwin-williams.com/dist/cvt/assets/images/collections/2019-shapeshifter-thumb.jpg',
        colorCollections: [
          '6060', '0059', '7625', '6404',
          '7743', '9169', '9086', '6116',
          '0079', '6049', '7735', '9006',
          '7640', '6369', '7551', '6992'
        ]
      }
    ]
  }
]
export const getColorCollectionsData = (colors: Color, allCollectionsData: any, tabId: string) => {
  let result = []
  let collection = allCollectionsData.find((data) => {
    return data.tabId === tabId
  })
  // temporarily added, following if condition should be removed once all collections data is available
  if (!collection) {
    collection = allCollectionsData.find((data) => {
      return data.tabId === 'tab1'
    })
  }
  collection && collection.collections.forEach((data) => {
    let tmp = {
      name: data.name,
      img: data.img,
      collections: []
    }
    for (let i = 0; i < data.colorCollections.length; i++) {
      const tmpColor = Object.values(colors).find((color) => {
        return color.colorNumber === data.colorCollections[i]
      })
      tmp.collections.push(tmpColor)
    }
    result.push(tmp)
  })
  return result
}

export const colorCollectionsData = getColorCollectionsData(colors, allCollectionsData, 'tab1')
export const expertColorsData = getExpertColors(colors, colorList)
export const collectionTabs = [
  { id: 'tab1', tabName: 'Most Popular' },
  { id: 'tab2', tabName: 'Color ID' },
  { id: 'tab3', tabName: 'Our Finest Whites' },
  { id: 'tab4', tabName: 'Color Forecast' },
  { id: 'tab5', tabName: 'Pottery Barn' },
  { id: 'tab6', tabName: 'West Elm' },
  { id: 'tab7', tabName: 'Lifestyle' },
  { id: 'tab8', tabName: `Kids' Colors` }
]
