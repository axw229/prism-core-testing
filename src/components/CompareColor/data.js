export const Scene = {
  'id': 1,
  'width': 1311,
  'height': 792,
  'variant_names': [
    'day',
    'night'
  ],
  'variants': [
    {
      'name': 'Living Room Day',
      'variant_name': 'day',
      'image': 'https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_livingroom6_day?wid=1311}&qlt=92',
      'thumb': 'https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_livingroom6_day?wid=300}&qlt=80',
      'normalizedImageValueCurve': '0 0.1 0.15 0.25 0.4 0.5 0.6 0.7 0.84 0.955 1',
      'surfaces': [
        {
          'id': 1,
          'mask': 'https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_livingroom6_day?wid=1280&req=object&opac=100&fmt=png8&object=wall&color=000000}&fmt=png8&bgColor=FFFFFF&op_invert=1',
          'hitArea': '/prism/images/scenes/rooms/2/m1.svg',
          'role': 'main'
        },
        {
          'id': 2,
          'mask': 'https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_livingroom6_day?wid=1280&req=object&opac=100&fmt=png8&object=accent&color=000000}&fmt=png8&bgColor=FFFFFF&op_invert=1',
          'hitArea': '/prism/images/scenes/rooms/2/m2.svg',
          'role': 'main'
        },
        {
          'id': 3,
          'mask': 'https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_livingroom6_day?wid=1280&req=object&opac=100&fmt=png8&object=trim&color=000000}&fmt=png8&bgColor=FFFFFF&op_invert=1',
          'hitArea': '/prism/images/scenes/rooms/2/m3.svg',
          'role': 'trim'
        }
      ]
    },
    {
      'name': 'Living Room Night',
      'variant_name': 'night',
      'image': 'https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_livingroom6_night?wid=1311}&qlt=92',
      'thumb': 'https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_livingroom6_night?wid=300}&qlt=80',
      'normalizedImageValueCurve': '0 0.1 0.22 0.32 0.42 0.51 0.6 0.7 0.84 0.955 1',
      'surfaces': [
        {
          'id': 1,
          'mask': 'https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_livingroom6_night?wid=1280&req=object&opac=100&fmt=png8&object=wall&color=000000}&fmt=png8&bgColor=FFFFFF&op_invert=1',
          'hitArea': '/prism/images/scenes/rooms/2/m1.svg',
          'role': 'main'
        },
        {
          'id': 2,
          'mask': 'https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_livingroom6_night?wid=1280&req=object&opac=100&fmt=png8&object=accent&color=000000}&fmt=png8&bgColor=FFFFFF&op_invert=1',
          'hitArea': '/prism/images/scenes/rooms/2/m2.svg',
          'role': 'main'
        },
        {
          'id': 3,
          'mask': 'https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_livingroom6_night?wid=1280&req=object&opac=100&fmt=png8&object=trim&color=000000}&fmt=png8&bgColor=FFFFFF&op_invert=1',
          'hitArea': '/prism/images/scenes/rooms/2/m3.svg',
          'role': 'trim'
        }
      ]
    }
  ]
}
