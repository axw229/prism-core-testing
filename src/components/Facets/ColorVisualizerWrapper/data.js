export const renderingData = {
  color: {
    content: {
      title: 'Explore all the ways to start building My Color Palette.',
      subContent: [
        {
          image: '/src/images/color-visualizer-wrapper/color-submenu__thumbnail--explore-color.png',
          subTitle: 'DIGITAL COLOR WALL',
          subContent: 'Search Sherwin-Williams paint colors to add to your palette or get color info.',
          url: '/active/color-wall/section/sherwin-williams-colors'
        },
        {
          image: '/src/images/color-visualizer-wrapper/color-submenu__thumbnail--color-collections.png',
          subTitle: 'COLOR COLLECTIONS',
          subContent: 'Find colors that work together seamlessly throughout your home.',
          url: '/color-collections'
        },
        {
          image: '/src/images/color-visualizer-wrapper/color-submenu__thumbnail--match-photo.png',
          subTitle: 'MATCH A PHOTO',
          subContent: 'Capture colors from any photo and match them to real paint colors.',
          url: '/match-photo'
        }
      ]
    }
  },
  inspiration: {
    content: {
      title: 'Find inspiration for your next painting project.',
      subContent: [
        {
          image: '/src/images/color-visualizer-wrapper/inspiration-submenu__thumbnail--painted-scenes.png',
          subTitle: 'PAINTED PHOTOS',
          subContent: 'See colors in context to start building your color confidence.',
          url: '/color-from-image'
        },
        {
          image: '/src/images/color-visualizer-wrapper/inspiration-submenu__thumbnail--expert-color-picks.png',
          subTitle: 'EXPERT COLOR PICKS',
          subContent: 'Find expertly designed color palettes to use in your space.',
          url: '/expert-colors'
        },
        {
          image: '/src/images/color-visualizer-wrapper/inspiration-submenu__thumbnail--sample-photos.png',
          subTitle: 'INSPIRATIONAL PHOTOS',
          subContent: 'Use our images to discover colors from around the world.',
          url: '/color-from-image'
        }
      ]
    }
  },
  scenes: {
    content: {
      title: 'Start Painting.',
      subContent: [
        {
          image: '/src/images/color-visualizer-wrapper/scene-submenu__thumbnail--sample-scenes.png',
          subTitle: 'USE OUR PHOTOS',
          subContent: 'Virtually paint a sample photo.',
          url: '/paint-scene'
        },
        {
          image: '/src/images/color-visualizer-wrapper/scene-submenu__thumbnail--my-photos.png',
          subTitle: 'UPLOAD YOUR PHOTO',
          subContent: 'Virtually paint your own photo.',
          description: 'Photo Tips: For best results make sure your room is well lit. Natural light works best.',
          url: '/paint-scene'
        }
      ]
    }
  }
}
