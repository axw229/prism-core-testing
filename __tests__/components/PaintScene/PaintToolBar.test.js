import React from 'react'
import { shallow } from 'enzyme'
import PaintToolBar from 'src/components/PaintScene/PaintToolBar'
import { toolNames } from 'src/components/PaintScene/data'

const defaultProps = {
  activeTool: toolNames.PAINTAREA,
  setActiveTool: jest.fn(),
  clearCanvas: jest.fn(),
  paintBrushShape: 'round',
  paintBrushWidth: 38,
  eraseBrushShape: 'round',
  eraseBrushWidth: 38,
  setBrushShapeSize: jest.fn(),
  performUndo: jest.fn(),
  performRedo: jest.fn(),
  undoIsEnabled: true,
  redoIsEnabled: true
}

const createPaintToolBarComponent = (props) => {
  return shallow(<PaintToolBar {...defaultProps} {...props} />)
}

/* @todo review these tests, I think more nuance is needed.
describe('PaintToolBar render testing', () => {
  it('Component should show brush palette when paint brush tool is selected and clicked', () => {
    const wrapper = createPaintToolBarComponent({ activeTool: toolNames.PAINTBRUSH })
    const mockEvent = { target: {}, preventDefault: jest.fn(), stopPropagation: jest.fn() }
    wrapper.find('button').at(1).simulate('click', mockEvent)
    const brushPalette = wrapper.find('.paint-tool-bar__brush-types--show').at(0).exists()

    expect(brushPalette).toBeTruthy()
  })

  it('Component should not show brush palette when paint brush tool is first selected', () => {
    const wrapper = createPaintToolBarComponent({ activeTool: toolNames.PAINTAREA })
    const mockEvent = { target: {}, preventDefault: jest.fn(), stopPropagation: jest.fn() }
    wrapper.find('button').at(0).simulate('click', mockEvent)
    const brushPalette = wrapper.find('.paint-tool-bar__brush-types--show').at(0).exists()

    expect(brushPalette).toBeFalsy()
  })

  it('Component should show erase palette when eraser tool is selected and clicked', () => {
    const wrapper = createPaintToolBarComponent({ activeTool: toolNames.ERASE })
    const mockEvent = { target: {}, preventDefault: jest.fn(), stopPropagation: jest.fn() }
    wrapper.find('button').at(1).simulate('click', mockEvent)
    const eraserPalette = wrapper.find('.paint-tool-bar__brush-types--show').at(0).exists()

    expect(eraserPalette).toBeTruthy()
  })

  it('Component should not show brush palette when paint brush tool is first selected', () => {
    const wrapper = createPaintToolBarComponent({ activeTool: toolNames.PAINTAREA })
    const mockEvent = { target: {}, preventDefault: jest.fn(), stopPropagation: jest.fn() }
    wrapper.find('button').at(1).simulate('click', mockEvent)
    const eraserPalette = wrapper.find('.paint-tool-bar__brush-types--show').at(0).exists()

    expect(eraserPalette).toBeFalsy()
  })
})
 */
