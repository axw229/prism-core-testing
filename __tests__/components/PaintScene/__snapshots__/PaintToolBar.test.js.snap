// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaintToolBar render testing should match snapshot 1`] = `
<Fragment>
  <div
    className="paint-tool-bar__group-tool paint-tool-bar__group-tool--hide"
  >
    <button
      className="paint-tool-bar__toolbar-button"
      key="1"
      name="deleteGroup"
      onClick={[Function]}
    >
      <FontAwesomeIcon
        border={false}
        className="paint-tool-bar__tool-icon"
        fixedWidth={false}
        flip={null}
        icon={
          Array [
            "fal",
            "trash-alt",
          ]
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size="lg"
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={
          Object {
            "rotate": 0,
          }
        }
      />
      <span
        className="paint-tool-bar__tool-name paint-tool-bar__tool-name--disabled "
      >
        DELETE
      </span>
    </button>
    <button
      className="paint-tool-bar__toolbar-button"
      key="2"
      name="group"
      onClick={[Function]}
    >
      <FontAwesomeIcon
        border={false}
        className="paint-tool-bar__tool-icon"
        fixedWidth={false}
        flip={null}
        icon={
          Array [
            "fal",
            "object-group",
          ]
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size="lg"
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={
          Object {
            "rotate": 0,
          }
        }
      />
      <span
        className="paint-tool-bar__tool-name paint-tool-bar__tool-name--disabled "
      >
        GROUP
      </span>
    </button>
    <button
      className="paint-tool-bar__toolbar-button"
      key="3"
      name="ungroup"
      onClick={[Function]}
    >
      <FontAwesomeIcon
        border={false}
        className="paint-tool-bar__tool-icon"
        fixedWidth={false}
        flip={null}
        icon={
          Array [
            "fal",
            "object-ungroup",
          ]
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size="lg"
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={
          Object {
            "rotate": 0,
          }
        }
      />
      <span
        className="paint-tool-bar__tool-name paint-tool-bar__tool-name--disabled "
      >
        UNGROUP
      </span>
    </button>
  </div>
  <div
    className="paint-tool-bar__wrapper"
  >
    <div
      className="paint-tool-bar__tools-container "
    >
      <button
        className="paint-tool-bar__toolbar-button paint-tool-bar__toolbar-button--active"
        key="1"
        name="paintArea"
        onClick={[Function]}
        onMouseDown={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className="paint-tool-bar__tool-icon"
          fixedWidth={false}
          flip="horizontal"
          icon={
            Array [
              "fa",
              "fill-drip",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="lg"
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={
            Object {
              "rotate": 0,
            }
          }
        />
        <span
          className="paint-tool-bar__tool-name paint-tool-bar__tool-name--active"
        >
          PAINT AREA
        </span>
      </button>
      <button
        className="paint-tool-bar__toolbar-button "
        key="2"
        name="paintBrush"
        onClick={[Function]}
        onMouseDown={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className="paint-tool-bar__tool-icon"
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "fa",
              "brush",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="lg"
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={
            Object {
              "rotate": 45,
            }
          }
        />
        <span
          className="paint-tool-bar__tool-name "
        >
          PAINTBRUSH
        </span>
      </button>
      <button
        className="paint-tool-bar__toolbar-button "
        key="3"
        name="selectArea"
        onClick={[Function]}
        onMouseDown={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className="paint-tool-bar__tool-icon"
          fixedWidth={false}
          flip="horizontal"
          icon={
            Array [
              "fa",
              "mouse-pointer",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="lg"
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={
            Object {
              "rotate": -20,
            }
          }
        />
        <span
          className="paint-tool-bar__tool-name "
        >
          SELECT
        </span>
      </button>
      <button
        className="paint-tool-bar__toolbar-button "
        key="4"
        name="erase"
        onClick={[Function]}
        onMouseDown={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className="paint-tool-bar__tool-icon"
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "fa",
              "eraser",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="lg"
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={
            Object {
              "rotate": 0,
            }
          }
        />
        <span
          className="paint-tool-bar__tool-name "
        >
          ERASE
        </span>
      </button>
      <button
        className="paint-tool-bar__toolbar-button "
        key="5"
        name="defineArea"
        onClick={[Function]}
        onMouseDown={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className="paint-tool-bar__tool-icon"
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "fal",
              "draw-polygon",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="lg"
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={
            Object {
              "rotate": 10,
            }
          }
        />
        <FontAwesomeIcon
          border={false}
          className="paint-tool-bar__tool-icon"
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "fal",
              "plus",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="xs"
          spin={false}
          style={
            Object {
              "right": "12px",
              "top": "21px",
            }
          }
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
        <span
          className="paint-tool-bar__tool-name "
        >
          DEFINE AREA
        </span>
      </button>
      <button
        className="paint-tool-bar__toolbar-button "
        key="6"
        name="removeArea"
        onClick={[Function]}
        onMouseDown={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className="paint-tool-bar__tool-icon"
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "fal",
              "draw-polygon",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="lg"
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={
            Object {
              "rotate": 10,
            }
          }
        />
        <FontAwesomeIcon
          border={false}
          className="paint-tool-bar__tool-icon"
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "fal",
              "minus",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="xs"
          spin={false}
          style={
            Object {
              "right": "12px",
              "top": "21px",
            }
          }
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
        <span
          className="paint-tool-bar__tool-name "
        >
          REMOVE AREA
        </span>
      </button>
      <button
        className="paint-tool-bar__toolbar-button "
        key="7"
        name="zoom"
        onClick={[Function]}
        onMouseDown={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className="paint-tool-bar__tool-icon"
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "fal",
              "search-plus",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="lg"
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={
            Object {
              "rotate": 0,
            }
          }
        />
        <span
          className="paint-tool-bar__tool-name "
        >
          ZOOM
        </span>
      </button>
      <button
        className="paint-tool-bar__toolbar-button "
        key="8"
        name="undo"
        onClick={[Function]}
        onMouseDown={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className="paint-tool-bar__tool-icon"
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "fa",
              "undo-alt",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="lg"
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={
            Object {
              "rotate": 0,
            }
          }
        />
        <span
          className="paint-tool-bar__tool-name "
        >
          UNDO
        </span>
      </button>
      <button
        className="paint-tool-bar__toolbar-button "
        key="9"
        name="redo"
        onClick={[Function]}
        onMouseDown={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className="paint-tool-bar__tool-icon"
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "fa",
              "redo-alt",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="lg"
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={
            Object {
              "rotate": 0,
            }
          }
        />
        <span
          className="paint-tool-bar__tool-name "
        >
          REDO
        </span>
      </button>
      <button
        className="paint-tool-bar__toolbar-button "
        key="10"
        name="hidePaint"
        onClick={[Function]}
        onMouseDown={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className="paint-tool-bar__tool-icon"
          fixedWidth={false}
          flip="horizontal"
          icon={
            Array [
              "fa",
              "eye",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="lg"
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={
            Object {
              "rotate": 0,
            }
          }
        />
        <span
          className="paint-tool-bar__tool-name "
        >
          HIDE PAINT
        </span>
      </button>
      <button
        className="paint-tool-bar__toolbar-button "
        key="11"
        name="info"
        onClick={[Function]}
        onMouseDown={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className="paint-tool-bar__tool-icon"
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "fal",
              "info-circle",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="lg"
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={
            Object {
              "rotate": 0,
            }
          }
        />
        <span
          className="paint-tool-bar__tool-name "
        >
          HINTS
        </span>
      </button>
      <div
        className="paint-tool-bar__brush-types paint-tool-bar__brush-types--hide paint-tool-bar__brush-types-paint paint-tool-bar__brush-types--show-by-opacity "
        onMouseLeave={[Function]}
      >
        <BrushTypes
          activeShape="round"
          activeWidth={38}
          hidePaintBrushTypes={[Function]}
          setBrushShapeSize={[MockFunction]}
        />
      </div>
      <div
        className="paint-tool-bar__brush-types paint-tool-bar__brush-types--hide paint-tool-bar__brush-types-erase paint-tool-bar__brush-types--show-by-opacity "
        onMouseLeave={[Function]}
      >
        <BrushTypes
          activeShape="round"
          activeWidth={38}
          hideEraseBrushTypes={[Function]}
          setBrushShapeSize={[MockFunction]}
        />
      </div>
      <div
        className="paint-tool-bar__zoom-tool paint-tool-bar__zoom-tool--hide "
      >
        <ZoomTool
          applyZoom={[MockFunction]}
        />
      </div>
    </div>
    <div
      className="paint-tool-bar__toolbar-toggle"
    >
      <button
        className="paint-tool-bar__toolbar-toggle-button paint-tool-bar__toolbar-toggle-button--toolbar-shown"
        onClick={[Function]}
      />
    </div>
  </div>
</Fragment>
`;
